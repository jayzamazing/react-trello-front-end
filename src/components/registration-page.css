.col-centered {
  float: none;
  margin: 0 auto;
}
.form-area {
  width: 340px;
  color: black;
}
