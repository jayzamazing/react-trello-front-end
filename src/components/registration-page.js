import React from 'react';
import {connect} from 'react-redux';
import {Link, Redirect} from 'react-router-dom';
import RegistrationForm from './registration-form';
import './registration-page.css';

export function RegistrationPage(props) {
    // If we are logged in (which happens automatically when registration
    // is successful) redirect to the user's dashboard
    if (props.loggedIn) {
        return <Redirect to="/boards" />;
    }
    return (
        <div className="registration container">
          <div className="row">
            <div className="col-centered form-area">
              <h2>Create a Trello Account</h2>
              <RegistrationForm />
              <p>Already have an account? <Link to="/login">Log in.</Link></p>
            </div>
          </div>
        </div>
    );
}

const mapStateToProps = state => ({
    loggedIn: state.auth.currentUser !== null
});

export default connect(mapStateToProps)(RegistrationPage);
