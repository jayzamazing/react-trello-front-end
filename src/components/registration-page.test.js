'use strict';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {RegistrationPage} from './registration-page';
import {Redirect} from 'react-router-dom';

describe('home-page', () => {
  it('should render page', () => {
    shallow(<RegistrationPage />);
  });
  it('snapshot', () => {
    const wrapper = shallow(<RegistrationPage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should redirect when user is logged in', () => {
    const wrapper = shallow(<RegistrationPage loggedIn="true"/>);
    expect(wrapper.find(Redirect)).toHaveLength(1);
  });
  it('redirect snapshow', () => {
    const wrapper = shallow(<RegistrationPage loggedIn="true"/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
