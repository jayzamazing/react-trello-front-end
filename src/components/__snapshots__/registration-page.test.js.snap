// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home-page redirect snapshow 1`] = `
<Redirect
  push={false}
  to="/boards"
/>
`;

exports[`home-page snapshot 1`] = `
<div
  className="registration container"
>
  <div
    className="row"
  >
    <div
      className="col-centered form-area"
    >
      <h2>
        Create a Trello Account
      </h2>
      <ReduxForm />
      <p>
        Already have an account? 
        <Link
          replace={false}
          to="/login"
        >
          Log in.
        </Link>
      </p>
    </div>
  </div>
</div>
`;
